src/routes/userRoute.ts

```diff
@@ -1,8 +1,9 @@
 import Router from 'express'
 
 import UserController from '../controllers/UserController'
+import signUpValidator from '../middlewares/userMiddleware'
 
 const router = Router()
-router.post('/signup', UserController.signup)
+router.post('/signup', signUpValidator, UserController.signup)
```

Comment:
```suggestion
router.post('/signup', validateRequest(userSchema, 'body'), UserController.signup)
```
We can use reusable  function instead, allowing use to define schema only 
```ts
router.post('/signup', [validateRequest(paramsSchema, 'params'), validateRequest(userSchema)], UserController.signup)
```